php-laravel-phan.md

```diff
@@ -0,0 +1,161 @@
+        'public',
+        'resources',
+        'routes',
+        'storage',
+        'tests',
+        'bootstrap',
+        'vendor/laravel',
+        'vendor/fakerphp'
+        'vendor/fruitcake',
+        'vendor/monolog',
+        'vendor/nesbot',
+        'vendor/ramsey',
+        'vendor/psr'
+        'vendor/symfony',        
+    ],
+
+    // A directory list that defines files that will be excluded
+    // from static analysis, but whose class and method
+    // information should be included.
+    //
+    // Generally, you'll want to include the directories for
+    // third-party code (such as "vendor/") in this list.
+    //
+    // n.b.: If you'd like to parse but not analyze 3rd
+    //       party code, directories containing that code
+    //       should be added to the `directory_list` as
+    //       to `exclude_analysis_directory_list`.
+    "exclude_analysis_directory_list" => [
+        'vendor'
+    ],
+
+    // A list of plugin files to execute.
+    // Plugins which are bundled with Phan can be added here by providing their name
+    // (e.g. 'AlwaysReturnPlugin')
+    //
+    // Documentation about available bundled plugins can be found
+    // at https://github.com/phan/phan/tree/v5/.phan/plugins
+    //
+    // Alternately, you can pass in the full path to a PHP file
+    // with the plugin's implementation.
+    // (e.g. 'vendor/phan/phan/.phan/plugins/AlwaysReturnPlugin.php')
+    'plugins' => [
+        // checks if a function, closure or method unconditionally returns.
+        // can also be written as 'vendor/phan/phan/.phan/plugins/AlwaysReturnPlugin.php'
+        'AlwaysReturnPlugin',
+        'DollarDollarPlugin',
+        'DuplicateArrayKeyPlugin',
+        'DuplicateExpressionPlugin',
+        'PregRegexCheckerPlugin',
+        'PrintfCheckerPlugin',
+        'SleepCheckerPlugin',
+        // Checks for syntactically unreachable statements in
+        // the global scope or function bodies.
+        'UnreachableCodePlugin',
+        'UseReturnValuePlugin',
+        'EmptyStatementListPlugin',
+        'LoopVariableReusePlugin',
+    ],
+    'suppress_issue_types' => [
+		// The following two have been added to not highlight issues
+		// raised due to variables added for interface methods that are
+		// then not used by the interface implementation
+		'PhanUnusedPublicMethodParameter',
+		'PhanUnusedProtectedMethodParameter',
+		'PhanUnusedPrivateMethodParameter',
+		// Allow unused values in foreach
+		'PhanUnusedVariableValueOfForeachWithKey'		
+	],
+	// Backwards Compatibility Checking
+	// (Disable this if the application no longer supports php 5,
+	// or use a different tool.
+	// Phan's checks are currently slow)
+	// Set it to false or omit it.
+	'backward_compatibility_checks' => false,
+	'unused_variable_detection' => true
+];
+
+```
+
+# Usage
+
+Add script command to run `phan` in `composer.json` file.
+
+```json
+"scripts": {
+    // ..<existing scripts>
+    "phan": "phan",
```

Comment:
Do you want to do something similar which you did with `phpcs`?

https://github.com/OsmosysSoftware/lint-docs/pull/21/files#diff-5ebe334af362550cb22dd6cc3db85d9e0c3eb108837c88538e9d8e601714bf2aR132

Reply:
Updated this to use the project level configuration.